### PR TITLE
Improve rollback by AMI functionality and clarity

### PR DESCRIPTION
**Pull request by macdiesel**
Please take a look! This will fix the rollback by AMI functionality and make it more clear.

### CODE REVIEW

File: tubular/asgard.py [hunk 1]
```diff
@@ -75,7 +78,7 @@ def clusters_for_asgs(asgs):
         u'test-edx-worker-v007',
     ]
 
-
+I want
```
Comment (feanil) [inline tubular/asgard.py]: Tell me what you want, what you really really want

**Review by feanil — APPROVED** [review]

File: tubular/asgard.py (hunk 1 above)
Comment (macdiesel) [reply tubular/asgard.py]: 🌶 :woman_singer: 🎸

File: tubular/asgard.py [hunk 2]
```diff
@@ -544,6 +548,16 @@ def rollback(current_clustered_asgs, rollback_to_clustered_asgs, ami_id=None):
                 LOG.info("Rollback ASG '{}' has been removed. Aborting rollback to ASGs.".format(asg))
                 rollback_ready = False
                 break
+    disabled_ami_id = None
+    try:
+        # fetch the currently deployed AMI_ID for the clusters for logging
+        if ami_id:
+            edp = ec2.edp_for_ami(ami_id)
+            disabled_ami_id = ec2.active_ami_for_edp(edp.environment, edp.deployment, edp.play)
+    except(MissingTagException, ImageNotFoundException, MultipleImagesFoundException):
```
Comment (doctoryes) [inline tubular/asgard.py]: Nit: Space after `except`.

File: tubular/asgard.py [hunk 3]
```diff
@@ -552,15 +566,27 @@ def rollback(current_clustered_asgs, rollback_to_clustered_asgs, ami_id=None):
...
         return deploy(ami_id)
     else:
         LOG.info("No AMI id specified - so no deploy occurred during rollback.")
-        return {'ami_id': None, 'current_asgs': current_clustered_asgs, 'disabled_asgs': rollback_to_clustered_asgs}
+        return {
+            'ami_id': None,
+            'current_ami_id': ami_id,
+            'current_asgs': current_clustered_asgs,
+            'disabled_ami_id': disabled_ami_id,
+            'disabled_asgs': rollback_to_clustered_asgs
```
Comment (doctoryes) [inline tubular/asgard.py:588]: Ugh - feel like I didn't get the names quite right initially.

File: tubular/asgard.py [hunk 4]
```diff
@@ -586,6 +614,9 @@ def deploy(ami_id):
     # Pull the EDP from the AMI ID
     edp = ec2.edp_for_ami(ami_id)
 
+    # fetch the active AMI_ID
+    disabled_ami_id = ec2.active_ami_for_edp(edp.environment, edp.deployment, edp.play)
```
Comment (doctoryes) [inline tubular/asgard.py]: Why didn't you add the try/except around this new call as well? Did you want the deploy to actually fail if this call failed?

Comment (doctoryes) [inline tubular/asgard.py]: I think all pipeline systems already have made this call back upstream to determine the base AMI on which a new AMI should be built. I wonder if that value could be used *or* maybe not used, but sanity-checked against.

**Review by doctoryes — COMMENTED** [review]
Looks good - just nits and questions.

File: tubular/asgard.py (hunk 4 above)
Comment (macdiesel) [reply tubular/asgard.py]: I think this is a good idea but out of scope for this fix.

Comment (macdiesel) [reply tubular/asgard.py]: Good point, I've added a try block around this.

**Comment by macdiesel** [comment]
Comments addressed.

**Review by doctoryes — APPROVED** [review]

**Pull request merged**